addon/models/course.js

```diff
@@ -1,271 +1,273 @@
         .filter((competency) => {
-          return competencies.includes(competency);
+          return courseCompetencies.includes(competency);
         })
         .sortBy('title');
 
-      return ObjectProxy.create({
-        content: domain,
+      return {
+        title: domain.title,
+        id: domain.id,
         subCompetencies,
-      });
+      };
     });
 
     return domainProxies.sortBy('title');
-  }),
-
-  requiredPublicationIssues: computed('startDate', 'endDate', 'cohorts.length', function () {
-    return this.getRequiredPublicationIssues();
-  }),
-  optionalPublicationIssues: computed(
-    'terms.length',
-    'courseObjectives.length',
-    'meshDescriptors.length',
-    function () {
-      return this.getOptionalPublicationIssues();
+  }
+
+  get requiredPublicationIssues() {
+    const issues = [];
+    if (!this.startDate) {
+      issues.push('startDate');
+    }
+    if (!this.endDate) {
+      issues.push('endDate');
     }
-  ),
 
-  /**
-   * All schools associated with this course.
-   * This includes the course-owning school, as well as schools owning associated cohorts.
-   * @property schools
-   * @type {Ember.computed}
-   * @public
-   */
-  schools: computed('school', 'cohorts.[]', async function () {
-    const courseOwningSchool = await this.school;
+    if (!this.cohorts.length) {
+      issues.push('cohorts');
+    }
 
-    const cohorts = await this.cohorts;
-    const programYears = await all(cohorts.mapBy('programYear'));
-    const programs = await all(programYears.mapBy('program'));
-    const schools = await all(programs.mapBy('school'));
+    return issues;
+  }
 
-    schools.pushObject(courseOwningSchool);
-    return schools.uniq();
-  }),
+  get optionalPublicationIssues() {
+    const issues = [];
+    if (!this.terms.length) {
+      issues.push('terms');
+    }
+    if (!this.courseObjectives.length) {
+      issues.push('courseObjectives');
+    }
+    if (!this.meshDescriptors.length) {
+      issues.push('meshDescriptors');
+    }
 
-  /**
-   * All vocabularies that are eligible for assignment to this course.
-   * @property assignableVocabularies
-   * @type {Ember.computed}
-   * @public
-   */
-  assignableVocabularies: computed('schools.@each.vocabularies', async function () {
-    const schools = await this.schools;
-    const vocabularies = await all(schools.mapBy('vocabularies'));
-    return vocabularies
-      .reduce((array, set) => {
-        array.pushObjects(set.toArray());
-        return array;
+    return issues;
+  }
+
+  @use _programYears = new ResolveAsyncValue(() => [this.cohorts.mapBy('programYear')]);
+  @use _programs = new ResolveAsyncValue(() => [this._programYears?.mapBy('program')]);
+  @use _programSchools = new ResolveAsyncValue(() => [this._programs?.mapBy('school')]);
+  @use _resolvedSchool = new ResolveAsyncValue(() => [this.school]);
+  get schools() {
+    if (!this._programSchools || !this._resolvedSchool) {
+      return undefined;
```

Comment:
why not return a blank array here instead of `undefined`? that would keep the output interface consistent.

Reply:
✅